Load dataset from Hugging Face and pass to UrbanMapper

soniacq opened the pull request:
## Loading Data from Hugging Face

This example loads the "oscur/pluto" dataset from Hugging Face, selects the training split, and converts the first 1,000 rows into a pandas DataFrame for efficient analysis and exploration. The resulting DataFrame can then be loaded into UrbanMapper using from_dataframe().

<img width="1079" alt="Screenshot 2025-04-25 at 4 02 01 PM" src="https://github.com/user-attachments/assets/42b2946f-9406-4d0f-88ac-3b51b9d9188c" />

simonprovost commented on examples/Basics/[1]loader.ipynb:
```diff
@@ -61,6 +61,52 @@
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Loading Data from Hugging Face\n",
+    "This code loads the \"oscur/pluto\" dataset from Hugging Face, selects the training split, and converts the first 1,000 rows into a pandas DataFrame for efficient analysis and exploration. The resulting DataFrame can then be loaded into UrbanMapper using from_dataframe()."
+   ]
+  },
+  {
```
Excellent!  Rapid modifications:

 1. Could we please reposition the `Hugging Face` cell after all the "more manual loaders"?
 2. We would also like to improve the introduction to this notebook, which should be mentioning two things now:
 -- That we can load locally available datasets in –– by default –– a manual way.
 -- That we also enable users to use the `HuggingFace dataset` library via `from_dataframe(.)`, which broadens the possibilities for integrating external data sources.
 
 At the same time, either in the introduction's markdown cell or the HF example's one, your call. We can use this opportunity to introduce the **OSCUR** `HuggingFace Dataset` source, (A) what it is about, and (B) explaining that it can be used in all subsequent notebook examples that require datasets to be used without issue because we have them all stored in.  I.e. demonstrating that there is no need to download the datasets from Google Drive or official links locally if not of interest –– we have them covered :) 
 
 EDIT 1.0: If introduction is changed, conclusion should be updated too, we mention 3 formats, it's actually four now, correct?

 Cheers

soniacq replied:
Fix in commit 692f0ec57c5973cf967fa61a059b28445e22f13f

simonprovost commented on examples/Basics/[1]loader.ipynb:
```diff
@@ -61,6 +61,52 @@
+    "dataset = load_dataset(\"oscur/pluto\")\n",
+    "# Select the training split\n",
+    "train_ds = dataset[\"train\"]\n",
+    "# Convert the first 1000 rows to a DataFrame\n",
+    "df = pd.DataFrame(train_ds[:1000])\n",
+    "df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "df_loader = (\n",
+    "    mapper\n",
+    "    .loader # From the loader module\n",
+    "    .from_dataframe(df) # To update with your dataframe\n",
+    "    .with_columns(longitude_column=\"longitude\", latitude_column=\"latitude\") # Inform your long and lat columns\n",
+    ")\n",
+    "\n",
+    "gdf = df_loader.load() # Load the data and create a geodataframe's instance\n",
+    "\n",
+    "# gdf stands for GeoDataFrame, like df in pandas for dataframes.\n",
+    "gdf"
+   ]
```
Not a big fan of having two cells for this. We can simply load data from HF (1), then UrbanMapper loading it using`from_dataframe` as you've done and showing the gdf at the end rather than showing it twice. It does not change.

```
# To update with your dataframe
```

👆 Unnecessary here.

soniacq replied:
Also fix with commit 692f0ec57c5973cf967fa61a059b28445e22f13f

simonprovost reviewed (changes requested):
Excellent! ☀️ That's gonna be highly highly useful throughout the UrbanMapper's workflow or any subsequent libraries coming out of OSCUR / VIDA! Thanks for this one.

Proposed some modifications. Finally, may you open a github issue's feature request, proposing the need for a `from_oscur(.)` / `from_huggingface_oscur(.)` throughout the `Loader` module to accelerate this process? While the feature request could or could never see the light of the day, at least it's written down and could be debated, with today's team and tomorrow's one.

Last suggestion, could you re work the commit's message please to follow Git Karma Convention: (1) https://karma-runner.github.io/6.4/dev/git-commit-msg.html, or you can look at the other commits we've done thus far (2) https://github.com/VIDA-NYU/UrbanMapper/commits/main/ ? Also explained [here](https://github.com/VIDA-NYU/UrbanMapper/blob/main/CONTRIBUTING.md) but as you know will soon be released in a proper documentation per #43 as soonish as I'm done with it ✅

Finally, one question, however; would you say we should include https://pypi.org/project/datasets/ within `UrbanMapper`? I would say _yes_ and ~no~. Yes because one of the example anyway's utilising it. No because we do not use it as part of the UrbanMapper's modules. What is your call here? If you believe it should be included, then you'll need to `uv add datasets` and commit the changes done to `pyproject.toml`.

Congrats for this PR! First contributor 🎉

simonprovost commented:
Hey @soniacq ,

Do not forget to rebase with `main` now that #43 and #46 are merged into the main track.  You may know how to do it, but in this case:

Make sure you have everything committed on you current work-based branch, including some dummy commits with names like `save` or any stash workflow.  Then proceed to `main` and `pull` the latest.  Git `checkout` back to your branch and run `git rebase main` to ensure you are "up to date" with `main`.  If conflicts arise, it is possible because the docs PR (#43) we completed with @ctsilva was quite large; determine what should be kept based on what is coming and your changes.  Feel free to message me if you are unsure, but you should be fine, trusting you on that it's no big deal ✅

To recap in action:

```
# Assuming you have nothing to commit
git checkout main
git pull
git checkout -
git rebase main
```

Tada 🎉

 Cheers!

soniacq commented:
@simonprovost  I have addressed all your comments—thank you for the valuable insights. I believe this PR is now ready to be merged into the master branch. Please have a look when you get a chance. Thanks!

simonprovost commented:
Thanks to @soniacq for this contrib! We needed this very much! Can't wait for #49 🎉
